client_example/25_wrapper/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ add_executable(client_tensor_transform_using_wrapper tensor_transform_using_wrap
 target_link_libraries(client_tensor_transform_using_wrapper PRIVATE composable_kernel::device_other_operations)
 add_executable(client_wrapper_img2col wrapper_img2col.cpp)
 target_link_libraries(client_wrapper_img2col PRIVATE composable_kernel::device_other_operations)
-if(GPU_TARGETS MATCHES "gfx9")
```

Comment:
FYI reviewers: this is not ideal :(

described here: https://github.com/ROCm/composable_kernel/issues/1371#issuecomment-2207020521
When we have a mixed multi-targets such as `-DGPU_TARGETS="gfx1100;gfx90a"`:
- Compiler will iterate over all named targets and try to build the client example 25. However, too many places directly uses XDL even inline, so this example should not be built with unsupported platforms;
- However, **if the goal is to build once, and run everywhere**:
    - **[Against what this change is doing]** we should still build this client-example, and at runtime check applicability and decide to skip or not;
    - However, IDK how to control compile time iterations, and skip certain lines/files based on which micro-architecture is been compiled for. e.g. `-DGPU_TARGETS="gfx1100;gfx90a"` means that compiler will compile for `--offload-arch=gfx1100 --offload-arch=gfx90a` in two iterations (and combine results together afterward). **is there any control to skip lines/files by MACRO or something similar in one of these compilation iterations?**

Reply:
This is attempting to dispatch compile time multi-targets properly:
https://github.com/ROCm/composable_kernel/pull/1372/commits/8ad46ed361b28362b59b13ecc4bd091fae9e7107